app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,13 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      all_options = ExtManagementSystem.leaf_subclasses.inject({}) do |ao, ems|
+        ao.merge(ems.ems_type => ems.respond_to?(:provider_settings) ? ems.provider_settings : {})
```

Comment:
Yes. I changed this to address `leaf_subclasses` of `ManageIQ::Providers::BaseManager` instead of `ExtManagementSystem` and it will be easy to add 
```
def self.provider_settings()
  {}
end
```
to this class.